clippy_lints/src/utils/conf.rs

```diff
@@ -277,6 +277,14 @@ define_Conf! {
     /// `".."` can be used as part of the list to indicate, that the configured values should be appended to the
     /// default configuration of Clippy. By default, any configuration will replace the default value.
     (disallowed_names: Vec<String> = super::DEFAULT_DISALLOWED_NAMES.iter().map(ToString::to_string).collect()),
+    /// Lint: SEMICOLON_INSIDE_BLOCK.
+    ///
+    /// Whether to lint only if it's multiline.
+    (semicolon_inside_block_if_multiline: bool = false),
+    /// Lint: SEMICOLON_OUTSIDE_BLOCK.
+    ///
+    /// Whether to lint only if it's singleline.
+    (semicolon_outside_block_if_singleline: bool = false),
```

Comment:
I changed the two to `semicolon_inside_block_ignore_singleline` and `semicolon_outside_block_ignore_multiline` since I agree, the fact that the names of both describe the effect is a bit weird.

You do understand the configuration - it ignores it if it's singleline/multiline. I thought to add two instead of only one as maybe someone may wish to always have a semicolon on the outside, but only inside if it's multiline, or some combination of the two.